firedrake/function.py

```diff
@@ -214,6 +177,125 @@ def __str__(self):
         else:
             return super(Function, self).__str__()
 
+    def as_coordinates(self):
+        """Create a mesh object using this function as coordinates."""
+        if hasattr(self, '_as_coordinates'):
+            mesh = self._as_coordinates()
+            if mesh is not None:
+                return mesh
+
+        from firedrake.mesh import MeshGeometry
+        return MeshGeometry(self)
```

Comment:
`as_coordinates` is a spectacularly misleading name.  Should this rather be a free function, that takes a Function and returns a mesh using that Function for its coordinate field?  `make_mesh_with_coordinates(function)`?